rust/processor/src/db_writer.rs

```diff
@@ -0,0 +1,98 @@
+/***
+   The goal of this module is to move db writes to their own executor pool
+   This allows us to have a separate pool of threads for db writes
+   In the future, we'd like to use a rayon pool for the core processing, and isolate async code
+*/
+
+use crate::{utils::database::PgDbPool, worker::PROCESSOR_SERVICE_TYPE};
+use diesel::QueryResult;
+use diesel_async::RunQueryDsl;
+use kanal::AsyncReceiver;
+use tokio::task::JoinHandle;
+use tracing::info;
+
+pub async fn launch_db_writer_tasks<
+    Query: diesel::query_builder::QueryFragment<diesel::pg::Pg>
+    + diesel::query_builder::QueryId
+    + Send
+    + 'static,
+>(
+    query_receiver: AsyncReceiver<Query>,
```

Comment:
So the query needs to be built before it can be sent to the `query_sender`. We can do that as part of the processor tasks since it's CPU-bound. Later we can think about splitting out query building as a separate task. 

So then the idea is that `process_transactions` will return the built + chunked queries. Then the queries get send to `query_sender` in worker.rs. Am I understanding that correctly?

Reply:
yep :-) exactly